pkg/source/resource_test.go

```diff
@@ -0,0 +1,160 @@
+					Name:      workloadName,
+				},
+			},
+			shouldError: false,
+			builder: resource.NewFakeBuilder(
+				func(version schema.GroupVersion) (resource.RESTClient, error) {
+					codec := k8sscheme.Codecs.LegacyCodec(scheme.PrioritizedVersionsAllGroups()...)
+					UnstructuredClient := &fake.RESTClient{
+						NegotiatedSerializer: resource.UnstructuredPlusDefaultContentConfig().NegotiatedSerializer,
+						Resp:                 &http.Response{StatusCode: http.StatusOK, Header: clitesting.DefaultHeader(), Body: clitesting.PodV1TableObjBody(codec, []client.Object{pod1Die})},
+					}
+					return UnstructuredClient, nil
+
+				},
+				c.ToRESTMapper,
+				func() (restmapper.CategoryExpander, error) {
+					return resource.FakeCategoryExpander, nil
+				},
+			),
+		},
+		{
+			name: "Fetch Empty Resource Object successfully",
+			args: []string{"pods"},
+			givenWorkload: &cartov1alpha1.Workload{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: defaultNamespace,
+					Name:      workloadName,
+				},
+			},
+			shouldError: false,
+			builder: resource.NewFakeBuilder(
+				func(version schema.GroupVersion) (resource.RESTClient, error) {
+					codec := k8sscheme.Codecs.LegacyCodec(scheme.PrioritizedVersionsAllGroups()...)
+					UnstructuredClient := &fake.RESTClient{
+						NegotiatedSerializer: resource.UnstructuredPlusDefaultContentConfig().NegotiatedSerializer,
+						Resp:                 &http.Response{StatusCode: http.StatusOK, Header: clitesting.DefaultHeader(), Body: clitesting.PodV1TableObjBody(codec, []client.Object{})},
+					}
+					return UnstructuredClient, nil
+
+				},
+				c.ToRESTMapper,
+				func() (restmapper.CategoryExpander, error) {
+					return resource.FakeCategoryExpander, nil
+				},
+			),
+		},
+		{
+			name: "Fetch  Resource Object Error",
+			args: []string{"pods"},
+			givenWorkload: &cartov1alpha1.Workload{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: defaultNamespace,
+					Name:      workloadName,
+				},
+			},
+			shouldError: true,
+			builder: resource.NewFakeBuilder(
+				func(version schema.GroupVersion) (resource.RESTClient, error) {
+					UnstructuredClient := &fake.RESTClient{
+						NegotiatedSerializer: resource.UnstructuredPlusDefaultContentConfig().NegotiatedSerializer,
+						Resp:                 &http.Response{StatusCode: http.StatusPreconditionFailed, Header: clitesting.DefaultHeader(), Body: nil},
+					}
+					return UnstructuredClient, nil
+
+				},
+				c.ToRESTMapper,
+				func() (restmapper.CategoryExpander, error) {
+					return resource.FakeCategoryExpander, nil
+				},
+			),
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			output := &bytes.Buffer{}
+			c.Stdout = output
+			c.Stderr = output
+			c.Builder = test.builder
+			labelparam := fmt.Sprintf("%s%s%s", cartov1alpha1.WorkloadLabelName, "=", test.givenWorkload.Name)
+			_, err := source.FetchResourceObjects(c.Builder, test.givenWorkload.Namespace, labelparam, test.args)
```

Comment:
update the test to assert on the expected table rows. 
```suggestion
			obj, err := source.FetchResourceObjects(c.Builder, test.givenWorkload.Namespace, labelparam, test.args)
			if obj != nil {
				gotTable := obj.(*metav1.Table)
				if d := cmp.Diff(gotTable.Rows, test.expected); d != "" {
					t.Errorf("Diff() %s", d)
				}
			}
```

on your test table struct
```
	expected       []metav1.TableRow
```